src/MinimumNodeService.cpp

```diff
@@ -57,20 +62,18 @@ void MinimumNodeService::setNormal(unsigned int nn)
 void MinimumNodeService::setUninitialised()
 {
   // DEBUG_SERIAL << F("> set Uninitialised") << endl;
-  requestingNewNN = false;
-  instantMode = MODE_UNINITIALISED;
   if (controller->getModuleConfig()->nodeNum != 0)
   {
     controller->sendMessageWithNN(OPC_NNREL);  // release node number first
   }
+  requestingNewNN = false;
+  instantMode = MODE_UNINITIALISED;
   controller->getModuleConfig()->setModuleUninitializedMode();
-  controller->getModuleConfig()->setCANID(0);
```

Comment:
MNS shouldn't really know about CANID. It is a configuration so belongs there. Having this code in configuration means that the code for dereferencing ( operator "->" ) is not needed.

Reply:
I agree.  CANID is also transport specific and so whilst it needs to be stored using Configuration, it may not be called that by other transports.  It might not even exist!  A thought for going forward as it is not an issue, yet.